### PR TITLE
Some FileManager extensions

### PR DESCRIPTION
**Pull request by rickpasetto**
In response to some feedback from https://github.com/tidepool-org/Loop/pull/247, I've consolidated some of the handier functions in a `FileManager` extension for use elsewhere.

(Note: this actually fixes a minor bug in `PersistenceController`: `fileExists` takes a _path_ not a _URL string_)

### CODE REVIEW

File: LoopKit/Persistence/PersistenceController.swift [hunk 1]
```diff
@@ -161,8 +159,7 @@ public final class PersistenceController {
                     options: [
                         NSMigratePersistentStoresAutomaticallyOption: true,
                         NSInferMappingModelAutomaticallyOption: true,
-                        // Data should be available on reboot before first unlock
-                        NSPersistentStoreFileProtectionKey: FileProtectionType.none
+                        NSPersistentStoreFileProtectionKey: FileProtectionType.completeUntilFirstUserAuthentication
```
Comment (rickpasetto) [inline LoopKit/Persistence/PersistenceController.swift:193]: This also changes the default data protection of the .sqlite file.  After perusing the documentation, I'm not sure what CoreData does if you change the `NSPersistentStoreFileProtectionKey`.  One hopes it does something like the extension above does.

Comment (ps2) [reply LoopKit/Persistence/PersistenceController.swift:193]: Yeah, I'm not sure. If we really wanted to migrate the protection levels of existing files, I'm guessing we'd have to do more than just this.

**Review by ps2 — APPROVED** [review]
LGTM. This needs testing, for sure, with existing db files.

File: LoopKit/Persistence/PersistenceController.swift (hunk 1 above)
Comment (rickpasetto) [reply LoopKit/Persistence/PersistenceController.swift:193]: @ps2 Maybe.  This SO post (admittedly it's weak evidence) seems to indicate that CoreData handles this:

https://stackoverflow.com/questions/36450555/changing-nspersistentstorefileprotectionkey-for-persistent-store-manager-on-exis

Would you like me to double-check before merging this?  It'd be easiest to just forge a new app and test it.

Comment (rickpasetto) [reply LoopKit/Persistence/PersistenceController.swift:193]: @ps2 I have written a sample app to test the theory about whether CoreData automatically updates the file protection mode, and I have convinced myself it does.  If you want details, I'd be happy to share them.

**Pull request merged**